### PR TITLE
GH-724 Nested Tables

### PR DESCRIPTION
**Pull request by GordonSmith**
Add support for DDL defined nested widgets inside a table.

Fixes GH-724

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage @mlzummo Please Review

**Comment by jbrundage** [comment]
@GordonSmith Could you add a link to a dermatology.html test case for your pull requests? (where applicable)

**Comment by GordonSmith** [comment]
ok, need to rebase (again) as well...

File: gulpfile.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ const cfg = {
 };
 
 const libs = ["d3", "c3", "colorbrewer", "dagre", "topojson", "d3-cloud", "font-awesome", "amcharts", "es6-promise", "amcharts.funnel", "amcharts.gauge", "amcharts.pie", "amcharts.radar", "amcharts.serial", "amcharts.xy", "amcharts.gantt", "amcharts.plugins.responsive", "simpleheat", "d3-hexbin", "d3-tip", "d3-bullet", "d3-sankey"];
-const bundles = ["common", "layout", "api", "other", "chart", "form", "c3chart", "google", "amchart", "tree", "graph", "map", "marshaller", "composite"];  //  Order is important ---
+const bundles = ["common", "layout", "api", "other", "chart", "form", "c3chart", "google", "amchart", "tree", "graph", "map", "composite", "marshaller"];  //  Order is important ---
```
Comment (jchambers-ln) [inline gulpfile.js]: since you have "//order is important" and this is changing the order is that intended?

File: src/marshaller/HipieDDL.js [hunk 2]
```diff
@@ -902,7 +945,47 @@
     Visualization.prototype.notify = function () {
         if (this.source.hasData()) {
             if (this.widget) {
+                var columnsRHS = this.source.getColumnsRHS();
                 var data = this.source.getData();
+                if (data instanceof Array) {
+                    //  TODO:  Move into source.getData  ---
```
Comment (jchambers-ln) [inline src/marshaller/HipieDDL.js]: todo still here

**Comment by jchambers-ln** [comment]
see in line this one is quite complex but I didn't see anything that jumped out as real bad -- one thing you are doing a lot of comping against inline strings may want to consider a include with all the constants such as "CHORO" to centralize it?  but that is subjective in JS as to which one is better.

**Pull request merged**